doc/contributing.rst

```diff
@@ -412,12 +412,135 @@ To avoid accidental highlighting follow this rule in ``*.nim`` files:
+separate github pages and reading through multiple comments in order to
+figure out what was *wrong* - not what *changed* in the commit.
+
+Instead, commit title should contain a human-readable text (we are
+optimizing for humans after all)
+
+- Keep the title length to 50 characters, and commit message body to 72
+  characters.
+- If you want to use automatic "fixes" feature of the github, please supply
+  full URL in the body instead, it makes it much easier to copy-paste and
+  see (with pure hash you first need to understand whether change was in
+  ``issues/`` or ``pull/``, then open random PR/issue and replace hash
+  URL).
+- Consider using "Fixes" for commits that fix bugs, and "Closes" for
+  commits that implement features.
+- Most of the points on this list were derived from `How To Write a Good
+  Commit Message
+  <https://api.coala.io/en/latest/Developers/Writing_Good_Commits.html>`_ -
+  we suggest reading it as well.
+
+Crafting good commit title might be hard at first. Some simple rules to
+look out for:
+
+- Avoid using phrases like ``"minor fixes for XXX"``, ``"various
+  improvements"``, ``"tiny patch"`` - they don't add any substantial
+  information and only clutter the title.
+- No bare github hashes in the title. Things like ``fix #19193 (#19195)
+  [backport:1.2]`` are absolutely illegible.
+
+When writing commit message consider who are you writing this message
+for:
+
+- you or someone else in the far future found a big and stumbled upon
+  this in a git bisect
+- you or someone else is going through recent commits trying to write a
+  changelog
+- you or someone else is trying to find a bigt that can't be replicated
+  reliably so they're reading commits seeing which might be related
+- someone new to the code base is trying to learn bits and pieces about
+  what's being worked on and what types of things are changing
+- you or someone else is sorting out documentation or test gaps, or
+  seeing if old issues can be closed with recent changes
+
+
+Message examples
+<<<<<<<<<<<<<<<<
+
+This section is build in a do-don't manner.
+
+- Implementation details of a particular commit can go into the body, title
+  should only contain *what?*
+  - ``fixes #9456 by only calling c_fclose if non nil`` -> ``Fix crash when
+    closing an unopened file on debian``
+    - Implementation details of the commit can be put in the commit body
+  - ``Deprecate 'c', 'C' prefix for octal literals, fixes #8082 (#8178)``
+    - Good, but commit hash should preferrably be replaced with a url.
+  - ``fixes #12015 by also checking kind of typeNode`` -> ``Fix unmarshal
+    of aliased tuple field using json.to``
+    - Again, implementain details go into commit body
+- No bare commit hashes:
+  - ``fixes #19051 [backport:1.6] (#19133)`` -> ``Fix gcc codegen of `type
+    G[T] = int``` or ``Fix gcc codegen of generic int alias``. Pr **body**
+    should be
+
+    .. code::
+
+       Fixes https://github.com/nim-lang/Nim/issues/19051
```

Comment:
```suggestion
       Fixes https://github.com/nim-works/nimskull/issues/19051
```

Reply:
That one should retain the original url, we don't have a 19k issues at the moment